brick-editor.js

```diff
@@ -35,163 +37,487 @@ function addBlocksHTML() { // eslint-disable-line no-unused-vars
                     }
-                } 
+                }
             }
-            if (numNodesCommonParent == positions.length) {
+            if (numNodesCommonParent == cursors.length) {
                 parentNode = node;
             }
         }
     });
-    // if no parentNode found, then position is after last character and parentNode = "Program"
+    // if no parentNode found, then cursor is after last character and parentNode = "Program"
     if (parentNode == null) {
         parentNode = ast.program;
     }
     return parentNode;
 }
 
 /**
- * Find the closest parent node that contains the position.
+ * Find the closest parent node that contains the cursor.
  *
- * @param {AST} ast - the root of the AST to search through.
- * @param {Location} position - A LineNumber and Column object.
- * @returns {node} - The AST node of the parent.
+ * @param {AST} ast - The root of the AST to search through.
+ * @param {Cursor} cursor - The line and column of the cursor.
+ * @returns {AST} - The AST node of the parent.
  */
-function findClosestParent(ast, position) {
-    return findClosestCommonParent(ast, [position]);
+function findClosestParent(ast, cursor) {
+    return findClosestCommonParent(ast, [cursor]);
 }
 
 /**
- * Find the immediately previous sibling to the position.
+ * Find the closest parent node that is able to be deleted.
  *
- * @param {AST} ast - the root of the AST to search through.
- * @param {Location} position - A lineNumber and column object.
+ * @param {AST} ast - The root of the AST to search through.
+ * @param {[Cursor]} cursors - A list of cursors.
+ * @returns {AST} - The AST node of the sibling.
+ */
+function findClosestCommonDeletableBlock(ast, cursors) {
+    var deleteNode = null;
+    estraverse.traverse(ast.program, {
+        enter: function (node) {
+            var numNodesCommonParent = 0;
+            for (var i = 0; i < cursors.length; i++) {
+                if (node.loc.start.line > cursors[i].lineNumber) {
+                    this.break();
+                }
+                if (node.loc.start.line <= cursors[i].lineNumber && node.loc.end.line >= cursors[i].lineNumber) {
+                    if ((node.type === "IfStatement" ||
+                        node.type === "ForStatement" ||
+                        node.type === "FunctionDeclaration" ||
+                        node.type === "WhileStatement" ||
+                        node.type === "ExpressionStatement" ||
+                        node.type === "ReturnStatement" ||
+                        node.type === "VariableDeclaration")) {
+                        if (node.loc.start.line == cursors[i].lineNumber) {
+                            if (node.loc.start.column <= cursors[i].column) {
+                                numNodesCommonParent++;
+                            }
+                        } else if (node.loc.end.line == cursors[i].lineNumber) {
+                            if (node.loc.end.column >= cursors[i].column) {
```

Comment:
Use `>==`

Reply:
Ditto for all other instances

Reply:
My editor doesn't seem to like the >==. May not be an issue though.

Reply:
...I might have made that up. Ignore me.